metashare/api/models.py

```diff
@@ -189,35 +200,119 @@ class Project(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Mode
     def __str__(self):
         return self.name
 
+    def subscribable_by(self, user):  # pragma: nocover
+        # TODO: revisit this?
+        return True
+
     def save(self, *args, **kwargs):
-        if not self.branch_name:
-            self.branch_name = slugify(self.name)
         super().save(*args, **kwargs)
 
+        if self.tracker.has_changed("branch_name"):
+            self.notify_has_branch_name()
+
+    def notify_has_branch_name(self):
+        from .serializers import ProjectSerializer
+
+        payload = ProjectSerializer(self).data
+        message = {"type": "PROJECT_UPDATE", "payload": payload}
+        async_to_sync(push.push_message_about_instance)(self, message)
+
     class Meta:
         ordering = ("-created_at", "name")
-        unique_together = (("name", "repository"), ("branch_name", "repository"))
+        unique_together = (("name", "repository"),)
 
 
 class TaskSlug(AbstractSlug):
     parent = models.ForeignKey("Task", on_delete=models.PROTECT, related_name="slugs")
 
 
 class Task(mixins.HashIdMixin, mixins.TimestampsMixin, SlugMixin, models.Model):
+    tracker = FieldTracker(fields=("branch_name",))
+
     name = models.CharField(max_length=50)
     project = models.ForeignKey(Project, on_delete=models.PROTECT, related_name="tasks")
     description = MarkdownField(blank=True, property_suffix="_markdown")
     assignee = models.ForeignKey(
         User, on_delete=models.SET_NULL, null=True, related_name="assigned_tasks"
     )
+    branch_name = models.SlugField(max_length=50, null=True)
 
     slug_class = TaskSlug
 
+    def __str__(self):
+        return self.name
+
+    def subscribable_by(self, user):  # pragma: nocover
+        # TODO: revisit this?
+        return True
```

Comment:
Right now this is only being used to live-update the `branch_url` while creating a scratch org... but I think it's okay?